nomenklatura/enrich/brightquery.py

```diff
@@ -0,0 +1,124 @@
+import os
+import requests
+import logging
+from banal import hash_data
+from typing import Generator, Optional
+from followthemoney.util import make_entity_id
+from followthemoney import DS, SE
+
+from nomenklatura.cache import Cache
+from nomenklatura.enrich.common import Enricher, EnricherConfig
+
+log = logging.getLogger(__name__)
+
+
+class BrightQueryEnricher(Enricher[DS]):
+    """Enricher for the BrightQuery Business Identity API."""
+
+    BASE_URL = "https://apigateway.brightquery.com/auth/business-identity-api/org"
+
+    def __init__(
+        self,
+        dataset: DS,
+        cache: Cache,
+        config: EnricherConfig,
+        session: Optional[requests.Session] = None,
+    ):
+        super().__init__(dataset, cache, config, session)
+
+        user = os.environ.get("BRIGHTQUERY_USER")
+        password = os.environ.get("BRIGHTQUERY_PASS")
+        if not user or not password:
+            raise ValueError(
+                "Missing BrightQuery credentials: BRIGHTQUERY_USER and/or BRIGHTQUERY_PASS"
+            )
+
+        self.session.auth = (user, password)
+
+    def match(self, entity: SE) -> Generator[SE, None, None]:
```

Comment:
it could be nice to split this into two methods
- one that does brightquery searches for an entity and yields results
- one that creates an entity per search result
match() can then call the second method on each result from the first method.

I know we don't split crawlers up this way, but we do that more with enrichers and I think it'd help with the readability of this long method.